docs/reference/types.md

```diff
@@ -0,0 +1,81 @@
+(types)=
```

Comment:
Moving this to reference feels like a great idea.

Reply:
I'd retitle to "Supported Go types and their format" and structure into:

## Supported Go types

SQLair supports the following types:

- in [input expressions]:
    - structs 
    - maps
    - slices
 - in [output expressions]:
    - structs
    - maps

[Note: If you want, you can link structs, maps, and slices to their Go definition too. To indicate to the user that the link is taking them outside of the present website, you can use a little caret, e.g., [structs^]...]

## Type format 

[everything you have now -- just nest the sections]

Reply:
Nice, done that